lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -923,57 +950,134 @@ def install_packages_from_local_dev_repo( host, package_name )
+            end
+
+            onhost_copied_file = File.join(onhost_copy_base, release_file)
+            fetch_http_file( release_path, release_file, copy_dir_local)
+            scp_to host, File.join(copy_dir_local, release_file), onhost_copy_base
+
+            case variant
+            when /^(fedora|el|centos|sles)$/
+              on host, "rpm -ivh #{onhost_copied_file}"
+            when /^(debian|ubuntu|cumulus)$/
+              on host, "dpkg -i --force-all #{onhost_copied_file}"
+              on host, "apt-get update"
+            when /^windows$/
+              result = on host, "echo #{onhost_copied_file}"
+              onhost_copied_file = result.raw_output.chomp
+              on host, Command.new("start /w #{onhost_copied_file}", [], { :cmdexe => true })
+            end
+            add_aio_defaults_on( host )
+          end
+        end
+
+        # Install shared repo of the puppet-agent on the given host(s).  Downloaded from 
+        # location of the form SHARE_BUILDS_URL/PE_VER/puppet-agent/AGENT_VERSION/repo
+        #
+        # @param [Host, Array<Host>, String, Symbol] hosts    One or more hosts to act upon,
+        #                            or a role (String or Symbol) that identifies one or more hosts.
+        # @param [Hash{Symbol=>String}] opts An options hash
+        # @option opts [String] :version The version of puppet-agent to install
+        # @option opts [String] :sha The sha of puppet-agent to install
+        # @option opts [String] :pe_ver The version of PE (will also use host['pe_ver']), defaults to '4.0'
+        # @option opts [String] :copy_base_local Directory where puppet-agent artifact
+        #                       will be stored locally
+        #                       (default: 'tmp/repo_configs')
+        # @option opts [String] :copy_dir_external Directory where puppet-agent
+        #                       artifact will be pushed to on the external machine
+        #                       (default: '/root')
+        # @option opts [String] :puppet_collection Defaults to 'PC1'
+        # @option opts [String] :share_builds_url Base URL to pull artifacts from
+        #
+        # @note on windows, the +:ruby_arch+ host parameter can determine in addition
+        # to other settings whether the 32 or 64bit install is used
+        #
+        # @example
+        #   install_puppet_agent_share_repo_on(host, { :sha => '1.1.0.227', :version => '1.1.0.227.g1d8334c', :pe_ver => '4.0.0-rc1'})
+        #
+        # @return nil
+        def install_puppet_agent_share_repo_on( hosts, opts )
+          block_on hosts do |host|
+            pe_ver = host[:pe_ver] || opts[:pe_ver] || '4.0'
+            variant, version, arch, codename = host['platform'].to_array
+            opts = FOSS_DEFAULT_DOWNLOAD_URLS.merge(opts)
+            opts[:download_url] = "#{opts[:share_builds_url]}/puppet-agent/#{pe_ver}/#{opts[:sha]}/repos/"
```

Comment:
this is just used as a part of the URL to fetch the install on L994, so as long as that works in the URL to the location of the install, this should be fine.  Is that right, @anodelman ?